src/docs/arch-connection.dox

```diff
@@ -0,0 +1,55 @@
+a time. Internally, a connection is represented by \c WT_CONNECTION.
+
+@section arch_conn_lifecycle Life cycle
+
+@subsection arch_conn_init Initialization
+
+A connection is initialized using WT_CONNECTION::wiredtiger_open. Given a set of configuration
+items (see @ref database_config), specific WiredTiger features can be enabled and their behavior
+tuned. Those features are for example related to @ref arch-eviction, @ref arch-logging, @ref
+arch-checkpoint, @ref arch-cache, statistics, etc. All the different available configuration
+settings are described in WT_CONNECTION::wiredtiger_open.
+
+WT_CONNECTION::wiredtiger_open also performs different sanity checks depending on the configuration
+item "create". When "create" is specified and a database does not exist already, a new database is
+created as well as specific WiredTiger files such as the turtle file and other metadata files. If a
+database already exists, WiredTiger will try to open it and check for the existence of the different
+required WiredTiger files. If "create" is not specified, WiredTiger expects a database where it is
+executed. In case of a failure during WT_CONNECTION::wiredtiger_open, the connection is not created
+and the returned error can be either \c WT_RUN_RECOVERY or \c WT_TRY_SALVAGE (see @ref
+error_handling) which means recovery might be required (see @ref command_line to learn more about
+it).
+
+Once the database has been successfully opened, worker threads are started to provide global
+services used during runtime. Those services consist of different threads to handle statistics,
+logging, eviction, checkpoint and cache management. The sweeping server that manages the active and
+inactive dhandles is started too, see @ref arch-dhandle for more information.
+
+Finally, before the connection is completely initialized, the database is set to a consistent state
+(see @ref arch-rts to read more about it).
+
+@subsection arch_conn_runtime Runtime
+
+During runtime, a number of actions can be executed using the connection. For instance, it is
+possible to reconfigure WiredTiger features and behavior using WT_CONNECTION::reconfigure instead of
+closing the connection and go through WT_CONNECTION::open again. Since any operation on the database
+can only be performed through a session, a connection can create one using
+WT_CONNECTION::open_session. The other available APIs are visible in the documentation related to
+WT_CONNECTION::.
+
+A connection also keeps tracks of global information, see \c WT_CONNECTION_IMPL defined in \c
+connection.h.
+
+@subsection arch_conn_closure Closure
+
+When a connection is no longer required, it can be closed using WT_CONNECTION::close. As a result,
+any resource held by the connection (i.e sessions) is freed unless configured differently and the
+database is left into a consistent state. It is worth noting that this final step might take some
```

Comment:
database is left into a consistent state -> database restored to a consistent state

Reply:
See 81955014b367084664e0a3e42283ff09a2839bac